xapian-core/weight/pl2weight.cc

```diff
@@ -70,7 +70,7 @@ PL2Weight::init(double)
     double base_change(1.0 / log(2.0));
     double mean = double(get_collection_freq()) / get_collection_size();
     P1 = mean * base_change + 0.5 * log2(2.0 * M_PI);
-    P2 = log2(mean) + base_change;
+    P2 = (mean>=(1/2.71828))?(log2(mean) + base_change):0;
```

Comment:
An inline comment here explaining what's going on might be helpful; it's not obvious just looking at it what this does.